src/patch-global.js

```diff
@@ -0,0 +1,7 @@
+if (!globalThis.URLPattern) {
+  const { URLPattern } = await import("./index.raw.js");
```

Comment:
Rather than dynamic import here I would import the implementation directly. App code or polyfill loaders can then dynamically load this file in a way that ensures their app code runs after the polyfill is loaded. Otherwise someone might import this and have to wait an unknown amount of time before using the URLPattern APIs.

Reply:
I personally think that's a bit of a separate concern. You have potentially three things:
1. Implementation
2. Patching the global
3. Dynamic loading the code

A lot of apps will have their own systems for (3) that ensure that app code is run after polyfills load. You could combine 2 and 3 here, but I do think you'd want to export the promise from the dynamic import so that app code could wait:

```ts
import {polyfillLoaded} from 'urlpattern-polyfill/loader.js';
await polyfillLoaded;
await import('./app.js');
```

My opinion is that there are lots of strategies for deciding what polyfills to load and when, so I'd leave that out. Some systems will want to bundle on the server based on UAs and make sure that there isn't a network waterfall after that, for instance.

So I'd make this file just:

```
import {URLPattern} from './url-pattern.js';
globalThis.URLPattern = URLPattern;
```

And that's it.

Reply:
So you wouldn't do the globalThis check either?

Reply:
OK pushed a new version, still doing the globalThis.URLPattern check